### PR TITLE
Query fix

### PR DESCRIPTION
**Pull request by Kaustavd**
Fix for issue #215

### CODE REVIEW

File: config/local.yml [hunk 1]
```diff
@@ -11,6 +11,8 @@ elasticsearch:
   - localhost
   cluster: foxtrot
   tableNamePrefix: foxtrot
+  countQueryTimeout: 10000
```
Comment (r0goyal) [inline config/local.yml]: Maybe create a map of opcode to timeout. That way adding timeout for other operations would be easier.

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/common/Action.java [hunk 2]
```diff
@@ -239,4 +240,11 @@ protected Filter getDefaultTimeSpan() {
         return filters;
     }
 
+    public long getCountQueryTimeout() {
```
Comment (r0goyal) [inline foxtrot-core/src/main/java/com/flipkart/foxtrot/core/common/Action.java]: Create one method which takes opcode as input and returns corresponding timeout.

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchConfig.java [hunk 3]
```diff
@@ -63,4 +66,12 @@ public String getTableNamePrefix() {
     public void setTableNamePrefix(String tableNamePrefix) {
         this.tableNamePrefix = tableNamePrefix;
     }
+
+    public long getCountQueryTimeout() {
+        return countQueryTimeout>0?countQueryTimeout:DEFAULT_TIMEOUT;
```
Comment (r0goyal) [inline foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchConfig.java]: format code.

File: config/local.yml (hunk 1 above)
Comment (Kaustavd) [reply config/local.yml]: opcode is understood by foxtrot. From ES query perspective there are largely two categorisation, aggregation and query_then_fetch. Hence I have kept two.

File: foxtrot-core/src/main/java/com/flipkart/foxtrot/core/common/Action.java (hunk 2 above)
Comment (Kaustavd) [reply foxtrot-core/src/main/java/com/flipkart/foxtrot/core/common/Action.java]: Does opcode make sense here as in ES query there is no significance of the same

**Comment by nitishgoyal13** [comment]
We already have addressed this in the commit (ef0737f). Hence, closing the MR

**Pull request closed without merging**